learn/get-started/work-with-data-using-queries-in-ballerina.md

```diff
@@ -161,14 +244,51 @@ public function findRecoveredPatientsOfCountries(table<CovidEntry> dataTable, st
 
-For every record in the `dataTable`, all the elements in the `countries` array will be joined. The output array of tuples will have the country and the number of recovered patients only if the condition after the `on` keyword is satisfied for that particular pair of table record and element of the array being joined.
+In this code:
+- The `findRecoveredPatientsOfCountries` function uses a query expression to join the `dataTable` table with an array of strings named `countries`. The join condition is provided after the `on` keyword. 
+- For every record in the `dataTable`, all the elements in the `countries` array will be joined. 
+- The output array of tuples will have the country and the number of recovered patients only if the condition after the `on` keyword is satisfied for that particular pair of table records and element of the array being joined.
+
+### Update the `main` function for joining
+
+To call the `findRecoveredPatientsOfCountries` function at the end to get the number of recovered patients, add the code below to the `main` function of the `main.bal` file. 
+
+>**Info:** You will get the number of recovered patients in the USA, India, and Afghanistan. 
+
+```ballerina
+    string[] c = ["USA", "India", "Afghanistan"];
+    [string, decimal][] countriesWithRecovered = findRecoveredPatientsOfCountries(covidTable, c);
+    io:println("Countries with number of Recovered patients:", countriesWithRecovered);
+```
+
+## The completed code for joining
 
-Now, change the existing `main` function by calling the `findRecoveredPatientsOfCountries` function at the end to get the number of recovered patients. 
-You will get the number of recovered patients in the USA, India, and Afghanistan. The updated `main` function should look like the one below.
+Below is the complete code for the joining function.
 
 ```ballerina
+public function filterCountriesByCases(table<CovidEntry> dataTable, decimal noOfCases) returns string[] {
+    string[] filteredCountries = from CovidEntry entry in dataTable
+        where entry.cases > noOfCases
+        select entry.country;
+    return filteredCountries;
+}
+
+public function findCountriesByHighestNoOfDeaths(table<CovidEntry> dataTable, int n) returns [string, decimal][] {
+    [string, decimal][] countriesWithDeaths = from CovidEntry entry in dataTable
+        order by entry.deaths descending
+        limit n
+        select [entry.country, entry.deaths];
+    return countriesWithDeaths;
+}
+
+public function findRecoveredPatientsOfCountries(table<CovidEntry> dataTable, string[] countries) returns [string, decimal][] {
+    [string, decimal][] countriesWithRecovered = from CovidEntry entry in dataTable
+        join string country in countries on entry.country equals country
+        select [entry.country, entry.recovered];
+    return countriesWithRecovered;
+}
```

Comment:
Commen in https://github.com/ballerina-platform/ballerina-dev-website/pull/4596/files#r917536655 applies to here also